Basic-Car-Maintenance/Shared/Settings/ViewModels/ChooseAppIconViewModel.swift

```diff
@@ -0,0 +1,40 @@
+import UIKit
+import Observation
+
+/// The ViewModel responsible for allowing users to change the AppIcon
+@Observable class ChooseAppIconViewModel {
+  private(set) var selectedAppIcon: AppIcon
+  
+  init() {
+    if let iconName = UIApplication.shared.alternateIconName,
+       let appIcon = AppIcon(rawValue: iconName) {
+      self._selectedAppIcon = appIcon
+    } else {
+      self._selectedAppIcon = .primary
+    }
+  }
+  
+  func updateAppIcon(to icon: AppIcon) {
+    let previousAppIcon = selectedAppIcon
+    selectedAppIcon = icon
+    
+    Task { @MainActor in
+      guard UIApplication.shared.alternateIconName != icon.iconName else {
+        print("No need to update icon since we're already using this icon.")
+        return
+      }
+      
+      do {
+        try await UIApplication.shared.setAlternateIconName(icon.iconName)
+        print("Successfully updated icon.")
```

Comment:
Can you remove the successful print statement please! It's fine to leave the "error" ones

And remove the one from line `23`, about not needing to update because that icon is already chosen. One thing you could add here is an error haptic, to still give the user feedback

Reply:
Good idea. Done.